homeworks/hw12/src/main.rs

```diff
@@ -0,0 +1,40 @@
+use bank_engine::bank::{Bank, BankTrait};
+
+fn main() {
+    let mut bank = Bank::new();
+
+    bank.create_account("Alice").unwrap();
+    bank.create_account("Bob").unwrap();
+
+    bank.deposit("Alice", 100.0).unwrap();
+    bank.withdraw("Alice", 50.0).unwrap();
+
+    bank.transfer("Alice", "Bob", 25.0).unwrap();
+
+    println!("Alice balance: {}", bank.get_balance("Alice").unwrap());
+    println!("Bob balance: {}", bank.get_balance("Bob").unwrap());
+
+    let history = bank.get_history();
+
+    if history.is_err() {
+        println!("History: {:?}", history.err().unwrap());
+        return;
+    }
+    let history = history.unwrap();
```

Comment:
```suggestion
    let Ok(history) = bank.get_history() else {
        println!("Error getting history: {:?}", history.err().unwrap());
        return;
    };
```

Так получается короче и логичнее, потому что не приходится дублировать проверку на ошибку, как это сделано сейчас с `is_err()` и `err().unwrap()`

Reply:
А вообще в функции `main()` немного намешаны стили обработки ошибок. То просто `unwrap()`, то логгирование и `return`. Притом, что в обоих случаях идет проверка на невозможную ошибку, т.е. не вижу логики в использовании разных стилей.

Предлагаю использовать какой-нибудь один стиль.
И для полной красоты можно еще посмотреть на крейты [`anyhow`](https://crates.io/crates/anyhow) и [`eyre`](https://crates.io/crates/eyre), с их помощью можно будет создать красивые ошибки для `main()` достаточно эргономично и без использования паники.